staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -902,6 +902,22 @@ const (
 	// Status code 500
 	StatusReasonServerTimeout StatusReason = "ServerTimeout"
 
+	// StatusReasonStoreReadError means that the server encountered an error while
+	// retrieving resources from the backend object store.
+	// This may be due to backend database error, or because processing of the read
+	// resource failed.
+	// Details:
+	//   "kind" string - the kind attribute of the resource being acted on.
+	//   "name" string - the prefix where the reading error(s) occurred
+	//   "causes" []StatusCause
+	//      - (optional):
+	//        - "type" CauseType - CauseTypeUnexpectedServerResponse
+	//        - "message" string - the error message from the store backend
+	//        - "field" string - the full path with the key of the resource that failed reading
+	//
```

Comment:
What would you think about also adding fields that could easily be copied to kubectl? Like for example:
- Resource = GVR
- ResourceName
That could be easier to use with kubectl. We could even merge both fields to make it simpler.

But at the end of the day it depends on which tool we want to use to delete resources. The key would be nice to have when using `etcdctl` and the full resource name would be more appropriate to `kubectl`

Reply:
I wonder if we can use the `TypeMeta` and `ListMeta` from the above-linked `StatusError`. This all happens pretty low in the stack and the data we have is quite limited.

Reply:
Maybe we could use StatusDetails no? https://github.com/kubernetes/kubernetes/blob/29f1cdbf995a6c6877d5694f90d73a209a42b148/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L785-L797

Reply:
Those are actually used here: https://github.com/kubernetes/kubernetes/pull/124674/files#diff-c6a5b875364aa1a724d30dde8081e897a5af7f24b09b48dbaf7cbf88455b5516R395-R398
though I'm not sure about the `Kind: qualifiedResource.Resource,` line

Reply:
Oh I see, I missed it when I had a second look. I was mostly going with the example you shared above:
```
    Details: &v1.StatusDetails{
      Name:   "/secrets",
      Group:  "",
      Kind:   "secrets",
      UID:    "",
```

It looks like we might want to fix the name value so that the actual name of the secret is exposed?

Reply:
Hmm, that's true. This error comes from listing, I think perhaps in that case the `Name` field should be omitted.

A get error currently looks like this:
```go
...
Details: &v1.StatusDetails{
  Name:   "/secrets/secret-encryption-test/pre-fail-c",
  Group:  "",
  Kind:   "secrets",
   UID:    "",
   Causes: []v1.StatusCause{
        v1.StatusCause{
           Type:    "UnexpectedServerResponse",
           Message: "cipher: message authentication failed",
           Field:   "/143d59bb-763f-4360-a86c-e7400375bfd6/registry/secrets/secret-encryption-test/pre-fail-c",
         },
    },
}
...
```
I suppose we could use the leaf element of the key instead of the full resource prefix in the storage. On the other hand, at least in the etcd store implementation, there is no concept of a "name" as that's not how the data is stored.

However, I've only literally just now noticed these [storage->API error transformators](https://github.com/kubernetes/kubernetes/blob/f307b0ae9d01c69256c2f72b7f50fd6022fc1941/staging/src/k8s.io/apiserver/pkg/storage/errors/storage.go#L40-L68) which I didn't see while writing the original enhancement.

I'll check if we could use these in this case to fill the status details with more relevant info.

Reply:
Maybe we could stick to the key in storage since it provides all the information that are needed to find the resource and just make it so that the utility to delete the resource can convert `/secrets/secret-encryption-test/pre-fail-c` into `/namespaces/secret-encryption-test/secrets/pre-fail-c`?

Or we could convert to an API path directly in the status so that it is also human readable

Reply:
I agree that sticking to the storage key and later converting it to the proper resource path makes most sense to me.

I'd rather use the raw data we have directly from the etcd client and do any modifications necessary for any further work with them in tooling later.